packages/realm-app-importer/src/AppImporter.ts

```diff
@@ -162,6 +168,30 @@ export class AppImporter {
     }
   }
 
+  private async fetch(url: FetchArgs[0], init?: FetchArgs[1]): Promise<FetchReturn> {
```

Comment:
```suggestion
  private async fetch(url: FetchArgs[0], init?: FetchArgs[1], retries = 0): Promise<FetchReturn> {
```

I would suggest implementing this as a recursive function with a `retries` counting up, to avoid having two calls to `fetch` in the same function body.